### ISSUE TITLE
Set operation must be inmutable

### ISSUE THREAD

**Issue by ivancorrales** [wesovilabs/koazee#22]
One of the principle of Koazee is that the elements are inmutable: It means that the input will never be modified by Koazee...

Set operations is modifying the input as it can be observed in this example.

```golang
package main

import (
	"fmt"
	"github.com/wesovilabs/koazee"
)

var numbers = []int{1, 5, 4, 3, 2, 7, 1, 8, 2, 3}

func main() {
	fmt.Printf("input: %v\n", numbers)

	stream := koazee.StreamOf(numbers)

	fmt.Print("stream.Set(0,5): ")
	fmt.Println(stream.Set(0, 5).Do().Out().Val())
	fmt.Printf("input: %v\n", stream.Out().Val())

}
```
input: [1 5 4 3 2 7 1 8 2 3]
stream.Set(0,5): [5 5 4 3 2 7 1 8 2 3]
input: [5 5 4 3 2 7 1 8 2 3]

This need to be fixed to release v0.0.3

**Comment by ivancorrales** [comment]
Fixed by PR #23

**Comment by ivancorrales** [comment]
Fixed
